### DISCUSSION TITLE
Idea: CLI tool for using rust-patterns as a manual

### DISCUSSION THREAD

**Discussion by simonsan** [rust-unofficial/patterns#240] [From Issues]
Something for the really far future:

Build a tool that pulls in this repository and wraps it `man`-like, so people could actually search within these articles from the command line. 

- distributable via `crate.io`
- `cargo install mdb-reader` to install, `cargo install-update mdb-reader` to update
- you can use the tool to search for patterns on the commandline working like `man` on Linux
- `mdb-reader rust-patterns builder` prints the `builder pattern` article in a nice commandline-native way
- nice to read something while developing in IDE
- could be used for other mdbooks as well, if there are multiple sources to pull data from, but would be the same basis

**Comment by marcoieni** [comment]
Basically it could be a way to read mdbooks interactively in the terminal, right?
I don't think the binary should be called `rust-patterns`. You install this general binary, that could be used for any book.
So it should be like `cargo install mdb-reader` or something like that.

**Comment by simonsan** [comment]
Updated the initial comment :-) Used the name because of this repository, but you're right in that it makes sense to chose a general name from the beginning to not let people get confused over this.

**Comment by pickfire** [comment]
Something like https://crates.io/crates/rusty-man? Or maybe we can open this in mdbook as an issue?
